polaris-react/src/components/Avatar/Avatar.tsx

```diff
@@ -95,24 +95,25 @@ export function Avatar({
     styles.Avatar,
     size && styles[variationName('size', size)],
     !customer && styles[variationName('style', styleClass(nameString))],
-    (hasImage || (initials && initials.length === 0)) &&
-      status !== Status.Loaded &&
-      styles.hidden,
-    hasImage && styles.hasImage,
+    hasImage && status === Status.Loaded && styles.imageHasLoaded,
+  );
+
+  const imageClassName = classNames(
+    styles.Image,
+    status !== Status.Loaded && styles.hidden,
   );
 
   const imageMarkUp =
     source && isAfterInitialMount && status !== Status.Errored ? (
       <Image
-        className={styles.Image}
+        className={imageClassName}
         source={source}
         alt=""
         role="presentation"
         onLoad={handleLoad}
         onError={handleError}
       />
     ) : null;
-
   // Use `dominant-baseline: central` instead of `dy` when Edge supports it.
   const verticalOffset = '0.35em';
```

Comment:
I played around with removing this in browserstack but it's still an issue in edge 101 😞